Remove internal use of deprecated configurations

ryanemerson opened the pull request:
https://issues.jboss.org/browse/ISPN-8300
https://issues.jboss.org/browse/ISPN-8294
https://issues.jboss.org/browse/ISPN-8308
https://issues.jboss.org/browse/ISPN-8313

wburns commented on core/src/test/java/org/infinispan/configuration/parsing/UnifiedXmlFileParsingTest.java:
```diff
@@ -523,11 +523,6 @@ private static void configurationCheck70(EmbeddedCacheManager cm) {
       assertTrue(c.dataContainer().<byte[]>keyEquivalence() instanceof AnyEquivalence);
       assertTrue(c.dataContainer().<byte[]>valueEquivalence() instanceof AnyEquivalence);
-
-      c = cm.getCacheConfiguration("store-as-binary");
```
We should verify that BINARY storage type has been enabled instead of just removing.

wburns commented on core/src/test/resources/configs/unified/9.0.xml:
```diff
@@ -198,14 +198,16 @@
             <property name="foo">bar</property>
          </data-container>
       </local-cache>
-      <local-cache name="store-as-binary">
```
This would should be converted to use `<memory> <binary/> <memory>` so it can pass the 70 test I mention above.

wburns commented on core/src/test/resources/configs/unified/9.1.xml:
```diff
@@ -198,14 +198,16 @@
             <property name="foo">bar</property>
          </data-container>
       </local-cache>
-      <local-cache name="store-as-binary">
```
Same as 9.0.xml above.

danberindei commented:
@wburns I'd also like to keep the deprecated attributes/elements in the input XMLs for now. 

@ryanemerson I guess we need a separate cache for each `memory` sub-element.

ryanemerson commented:
@wburns I've updated the Serializer to no longer serialize deprecated configurations where possible (data-container is still present) and updated the parser etc. Let me know what you think. I'm going to add an additional commit(s) which removes all erroneous uses of the EvictionConfiguration and StoreAsBinaryConfiguration

wburns commented on core/src/main/java/org/infinispan/configuration/parsing/Parser.java:
```diff
@@ -1870,33 +1865,22 @@ protected void parseTransaction(XMLExtendedStreamReader reader, ConfigurationBui
-            }
-            case MAX_ENTRIES: {
+            case MAX_ENTRIES:
                log.evictionMaxEntriesDeprecated();
```
We can remove this now too since all of eviction is deprecated.

ryanemerson replied:
@wburns Hmm, this has got me thinking. Can we make these changes to the parser for 9.1.x, or do we need to hold off this for 9.2.x?

wburns commented on core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java:
```diff
@@ -263,7 +261,7 @@ public void preload() {
 
 
-      final long maxEntries = getMaxEntries();
+      final long maxEntries = configuration.memory().size();
```
Hrmm this defaults to -1, should we not set this to `Long.MAX_VALUE` or ignore if it is -1?

wburns commented on hibernate-cache/src/main/java/org/infinispan/hibernate/cache/TypeOverrides.java:
```diff
@@ -60,6 +60,7 @@ public EvictionStrategy getEvictionStrategy() {
     *
     * @param evictionStrategy String defining eviction strategy allowed.
     *                         Possible values are defined in {@link EvictionStrategy}
+	 * @deprecated
```
Nitpick: whitespace

tristantarrant commented:
Merged, thanks
